src/gen/conditions.rs

```diff
@@ -258,6 +268,9 @@ pub fn parse_args(
             match sanitize_uint(a, first(a, c)?, 4, ErrorCode::AssertHeightRelative) {
                 // Height is always positive, so a negative requirement is always true,
                 Err(ValidationErr(_, ErrorCode::NegativeAmount)) => Ok(Condition::Skip),
+                Err(ValidationErr(n, ErrorCode::AmountExceedsMaximum)) => {
+                    Err(ValidationErr(n, ErrorCode::AssertHeightRelative))
```

Comment:
Is this preserving the current behavior? If so, let's leave it like this. Otherwise I'm tempted to suggest we should turn this into an `InvalidCondition`, because it *is* an invalid condition rather than a valid condition that failed.